src/processutils.ts

```diff
@@ -0,0 +1,94 @@
+import { spawn } from 'child_process'
+
+interface ProcessResult {
+  code: number
+  signal: number
+  stdout: string
+  stderr: string
+}
+
+interface Options {
+  detached?: boolean
+  env?: NodeJS.ProcessEnv
+  maxSize?: number
+  stdOutMaxSize?: number
+  stdErrMaxSize?: number
+  timeout?: number
+  closeStdin?: boolean
+}
+
+export async function runProcessAsync(command: string, args: string[], options: Options = {}): Promise<ProcessResult> {
+  const cmd = spawn(command, args, {
+    env: options.env,
+    detached: options.detached
+  })
+
+  const maxSize = options.maxSize || 10 * 1024 * 1024
+  const stdOutMaxSize = options.stdOutMaxSize || maxSize
+  const stdErrMaxSize = options.stdErrMaxSize || maxSize
+
+  const promise = new Promise((resolve, reject) => {
```

Comment:
```suggestion
  const promise = new Promise<ProcessResult>((resolve, reject) => {
```

If you do this, then the redundant piece in L66-L73 can be removed

Reply:
You committed my changes, but didn't remove the `any` in what is now L55

Reply:
and L67